ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py

```diff
@@ -3058,3 +3058,55 @@ def test_tensorflow_Elu(
         features=x[0],
         name=name,
     )
+
+
+@handle_frontend_test(
+    fn_tree="tensorflow.raw_ops.Conv2D",
+    x_f_d_df=_x_and_filters(
+        dtypes=helpers.get_dtypes("float", full=False),
+        data_format=st.sampled_from(["NHWC", "NCHW"]),
+        padding=st.sampled_from(["SAME", "VALID", "EXPLICIT"]),
+        type="2d",
+        dilation_min=1,
+        dilation_max=1,
+    ),
+    test_with_out=st.just(False),
+)
+def test_tensorflow_Conv2D(
+    *,
+    x_f_d_df,
+    test_flags,
+    frontend,
+    fn_tree,
+    on_device,
+):
+    input_dtype, x, filters, dilation, data_format, stride, padding = x_f_d_df
+
+    # Broadcast strides and dilations to correct dims for the ground truth
+    # backend func to run correctly
+    stride = _convolution_broadcast_helper(
+        stride, num_spatial_dims=2, channel_index=3, name="strides"
+    )
+    dilation = _convolution_broadcast_helper(
+        dilation, num_spatial_dims=2, channel_index=3, name="dilations"
+    )
+
+    if padding == "EXPLICIT" and data_format == "NHWC":
+        explicit_paddings = [(0, 0), (1, 1), (1, 1), (0, 0)]
+        padding = explicit_paddings
+    elif padding == "EXPLICIT" and data_format == "NCHW":
+        explicit_paddings = [(0, 0), (0, 0), (1, 1), (1, 1)]
```

Comment:
`explicit_paddings` should be randomly generated so that it's not always 1
you could have a look at how `ivy.conv2d`'s test generates it

Reply:
should i use the list helpers function or something else, I didn't understand it fully

Reply:
I think this will work
```
    explicit_paddings = draw(
            st.lists(
                st.tuples(
                    st.integers(min_value=0, max_value=3),
                    st.integers(min_value=0, max_value=3),
                ),
                min_size=2,
                max_size=2,
            )
    )
```
And then you'll have to append `(0,0)` depending on the `data_format`.
You'll need to use a `st.composite` strategy where you'll call `draw` the way I showed and then do the appending.